### PR TITLE
Mqtt v5 reason codes for connect, ack and disconnect messages

### PR DESCRIPTION
**Pull request by imbeacon**
## Pull Request description

Added Mqtt v5 result codes for mqtt v5 clients.

PE PR - https://github.com/thingsboard/thingsboard-pe/pull/1317

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.
  
## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.

### CODE REVIEW

**Review by YevhenBondarenko — APPROVED** [review]

File: application/src/test/java/org/thingsboard/server/transport/mqtt/mqttv3/credentials/BasicMqttCredentialsTest.java [hunk 1]
```diff
@@ -114,11 +115,15 @@ public void testCorrectCredentials() throws Exception {
...
     }
 
-    @Test(expected = MqttSecurityException.class)
+    @Test(expected = MqttException.class)
     public void testCorrectClientIdAndUserNameButWrongPassword() throws Exception {
         // Not correct. Correct clientId and username, but wrong password
         MqttTestClient mqttTestClient = new MqttTestClient(CLIENT_ID);
-        mqttTestClient.connectAndWait(USER_NAME3, "WRONG PASSWORD");
+        try {
+            mqttTestClient.connectAndWait(USER_NAME3, "WRONG PASSWORD");
```
Comment (ashvayka) [inline application/src/test/java/org/thingsboard/server/transport/mqtt/mqttv3/credentials/BasicMqttCredentialsTest.java:123]: Add Assert that checks we have not executed any code after connectAndWait and went straight to the catch block.

File: application/src/test/java/org/thingsboard/server/transport/mqtt/mqttv5/MqttV5TestClient.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+import org.eclipse.paho.mqttv5.client.MqttCallback;
+import org.eclipse.paho.mqttv5.client.MqttConnectionOptions;
+import org.eclipse.paho.mqttv5.client.persist.MemoryPersistence;
+import org.eclipse.paho.mqttv5.common.MqttException;
+import org.eclipse.paho.mqttv5.common.MqttMessage;
+import org.thingsboard.server.common.data.StringUtils;
+
+import java.util.concurrent.TimeUnit;
+
+public class MqttV5TestClient {
```
Comment (ashvayka) [inline application/src/test/java/org/thingsboard/server/transport/mqtt/mqttv5/MqttV5TestClient.java:30]: Add a note that we need a copy of MqttV3TestClient due do different package names in import.

File: common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java [hunk 3]
```diff
@@ -918,12 +940,24 @@ private static int getMinSupportedQos(MqttQoS reqQoS) {
...
-        MqttMessageIdVariableHeader mqttMsgIdVariableHeader =
-                MqttMessageIdVariableHeader.from(requestId);
-        return new MqttPubAckMessage(mqttFixedHeader, mqttMsgIdVariableHeader);
+    private static MqttVersion getMqttVersion(int versionCode) {
+        switch (versionCode) {
+            case 3:
+                return MqttVersion.MQTT_3_1;
+            case 5:
+                return MqttVersion.MQTT_5;
+            default:
```
Comment (ashvayka) [inline common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java:949]: "case 4:" is redundant.

File: common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/session/DeviceSessionCtx.java [hunk 4]
```diff
@@ -74,6 +75,12 @@ public class DeviceSessionCtx extends MqttDeviceAwareSessionContext {
     @Setter
     private boolean provisionOnly = false;
 
+    @Getter
+    @Setter
+    private MqttVersion mqttVersion;
+
+
```
Comment (ashvayka) [inline common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/session/DeviceSessionCtx.java:82]: remove redundant spaces.

**Pull request merged**